build.sh

```diff
@@ -197,6 +197,12 @@ if [ "$BUILD_TYPE" = "all" ]; then
     if [ $? -ne 0 ]; then exit 1; fi
 fi
 
+# Generate PlatformIO manifest file
+if [ "$BUILD_TYPE" = "all" ]; then
+    python3 ./tools/gen_platformio_manifest.py -o "$TOOLS_JSON_OUT/"
```

Comment:
I suggest you pass the branch and commit hash directly to the script. This way it will not depend on versions.txt which could get relocated, changed, etc. Keep in mind that sometimes the branch will return error and the tag will be taken, so string might be `v5.1.0` instead of `release/v5.1`. You have the option to split the arguments further and detect that condition for when releasing tags and not current branch versions
```sh
python3 ./tools/gen_platformio_manifest.py -o "$TOOLS_JSON_OUT/" -b $(git -C "$IDF_PATH" symbolic-ref --short HEAD || git -C "$IDF_PATH" tag --points-at HEAD) -c $(git -C "$IDF_PATH" rev-parse --short HEAD)
```

Reply:
Thanks for the hints, updated.